### ISSUE TITLE
[Feature Request] Port features from old json settings editor

### ISSUE THREAD

**Issue by lonix1** [microsoft/vscode#130457]
The "split JSON settings editor" was removed in 1.59.0. This [was apparently](https://github.com/microsoft/vscode/issues/125952#issue-917705621) done because it was old and hard-to-maintain code which represented significant tech debt to the team.

We've all been there - we understand!

Now that it has been replaced by the new editor - which is based on well-designed code that is easy to maintain - please port features that we sorely miss from the old editor:
- fuzzy search
- filtering
- pencil
- etc.

**Comment by CaselIT** [comment]
> fuzzy search

The fuzzy search would be a nice to have option also in the normal search that's used in everywhere

**Comment by Nick-Lucas** [comment]
This is a much more constructive approach than some others.✌️

I really empathise that this issue has brought out the worst in some community member's attitudes, and that makes it difficult for the core team to communicate on this issue, and may have brought out understandable stubbornness in the core team. But I also think it highlights just how loved and depended on the _full featureset_ of the settings JSON editing was by many long time and power users of Code.

It's really saddening that the stripping down of features is being marketed as a compromise, rather than finding a compromise which works for everyone, like:

* Commit to migrate to a new approach under the hood with the same experience - even if it takes a few weeks/months to come to fruition from now
* expose any missing extension APIs and spin the editor out into an extension which we as a community can help maintain

Just a couple ideas. Please could your team discuss some alternative strategies @roblourens ?

**Comment by NatoBoram** [comment]
I compiled a list of features that can be ported to the rest of the editors [here](https://github.com/microsoft/vscode/issues/125952#issuecomment-873151703), [here](https://github.com/microsoft/vscode/issues/125952#issuecomment-881134156) and [here](https://github.com/microsoft/vscode/issues/129594).

* Fuzzy search
* Hiding keys that aren't found by the search feature
* Searching in one panel should optionally search other panels, too
* Selecting a key in one panel should focus on that key on other panels
* The little pencil icon's "copy to settings" should be replaced by "copy to other panes"

**Comment by KerkDovan** [comment]
Don't know if this feature will be useful, but maybe it is possible to make automatic grouping and sorting of the overriden settings? With the possibility to turn it off, of course.

**Comment by lonix1** [comment]
May I suggest we refrain from adding a laundry list of new ideas to this request? If we do, there is zero chance it'll get implemented.

I think we need to plead our case as follows:
1. **Please add all the features that we lost**
2. Once we're back to normal, please consider making changes, additions, tweaks, etc.

Of course if they show willingness to consider new ideas as well, so much the better. But as of now they haven't even recognized this issue - even though it should already be on the backlog.

**Comment by KerkDovan** [comment]
Ok, that makes sense. I agree.

**Comment by Rory-Sullivan** [comment]
Happy to add my voice to this, I very much miss the old JSON settings editor. I have tried the GUI editor recently and it has improved a lot since the last time I used it but I still don't find it nearly as useful as the old JSON editor.

For me, the feature I miss the most is the searching functionality, everything else is not that much of a priority.

**Comment by amadeann** [comment]
Just opened the settings for the first time in a while, and saw the change. I installed a new extension and wanted to quickly scan the available settings for it. The extension's name starts with `xml.` In the old JSON editor, it was very easy to see ALL available customizations and identity the one I need. After the change I can either:

- do a regular search, switching on my "regex" brain to construct a good search phrase to lead me to the necessary section of the file. Fuzzy search allowed me to find settings easily, now I have to jump through hoops to get started in search.
- use the "UI" version of settings, which is pretty bad for getting an overview, as it requires me to scroll through multiple screenlenghts, because of "nice, human formatting", with huge spacing, dropdowns, etc. As a developer I don't care about the visuals, I am happy with a compact, console-like environment that is functional.

I also need to do "quick scanning" when I am debugging problems with extensions (e.g. autoformatting in JavaScript & HTML files comes to my mind right now, since it's something that often gets messed up).

In short: neither of the options after the changes is usable for my most common usecases. I would relly like to see the old JSON editor back.

**Comment by roblourens** [comment]
I think it's time to close this one. If there's something keeping you from being able to use the settings GUI, please open an issue.

**Comment by NatoBoram** [comment]
People who reacted to this change don't want to use the GUI. They want to use the JSON editor. And they want the old editor's features back. I don't get how someone can still be confused after it became [such a mess](https://github.com/microsoft/vscode/issues/129594).

* https://github.com/microsoft/vscode/issues/134190

**Comment by denisz1** [comment]
Close just before Christmas, so we away or not paying attention.

"In bad faith."

**Comment by gjsjohnmurray** [comment]
@denisz1 this House Keeping Iteration occurs annually at this time of year.

https://github.com/microsoft/vscode/wiki/Issue-Grooming

IMO it is an important part of this project sustaining an impressively regular cadence of 11 releases per year.

**Comment by migig** [comment]
Tagging top vscode contributors @bpasero, @jrieken, @mjbvz, @joaomoreno, @tyriar, before this is locked forever.

You insist we use the upvoting process. We did, and *upvoted this to one of the top issues, and in record time*. And this is not a feature request - it's a request to reimplement features removed while refactoring old code. And we did not harass you, we’ve been patient for years.

If you leave this on the backlog forever it would be upsetting, but that’s all. However, closing the issue, even though it has more upvotes than most of the 5,000+ issues, and has a very long and difficult history - which I'm sure you all know about - is both upsetting and very insulting.

Many of us do not use GUIs. We use file-based config which is easier to manage in our environments, and works better with git (and the the original issue has dozens of other use cases). I’m sure all of you are power users so you understand the issue.

**We followed your rules… please don’t make a mockery of your own processes.**

In the spirit of the season, please do the right thing and *leave this on the backlog*, or better yet, **engage with us**.

Thank you.
